src/components/Dashboard.tsx

```diff
@@ -106,9 +105,7 @@ const FilterComponent: React.FC<{ template: TemplateDetails; filter: Filter; set
   })
 
   useEffect(() => {
-    if (i+1 === filters.length) {
-    setLoading(loading)
-    }
+    if (loading === false) setLoadingFilters((loadingFilters: number) => loadingFilters + 1)
```

Comment:
As I mentioned above, you can remove "loadingFilters" from the props. So the "loadingFilters" value here is just the parameter for the function you're sending to `setLoadingFilters`, so can be any name.

I'd suggest: `setLoadingFilters((currentCount) => currentCount + 1)`